plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/router/MikamiTabuchiConnectionRouter.java

```diff
@@ -0,0 +1,284 @@
+        }
+    }
+
+    private void queueRerouting() {
+        if (this.connectionToPath != null && !this.connectionToPath.isEmpty()) {
+            try {
+                this.ignoreInvalidate = true;
+                ((Connection)this.connectionToPath.keySet().iterator().next()).revalidate();
+            } finally {
+                this.ignoreInvalidate = false;
+            }
+        }
+    }
+
+    void addChild(IFigure child) {
+        if (this.connectionToPath != null && !this.figuresToBounds.containsKey(child)) {
+            Rectangle bounds = child.getBounds().getCopy();
+            this.algorithm.addObstacle(bounds);
+            this.figuresToBounds.put(child, bounds);
+            child.addFigureListener(this.figureListener);
+            this.isDirty = true;
+
+        }
+    }
+
+    public MikamiTabuchiConnectionRouter(IFigure container) {
+        this.container = container;
+    }
+
+    private void processStaleConnections() {
+        Iterator<Connection> iter = this.staleConnections.iterator();
+        if (iter.hasNext() && this.connectionToPath == null) {
+            this.connectionToPath = new HashMap<>();
+            this.hookAll();
+        }
+
+        OrthogonalPath path;
+        for(; iter.hasNext(); this.isDirty |= path.isDirty()) {
+            Connection conn = iter.next();
+            path = this.connectionToPath.get(conn);
+            if (path == null) {
+                path = new OrthogonalPath(conn);
+                this.connectionToPath.put(conn, path);
+                this.algorithm.addPath(path);
+            }
+
+            List constraint = (List) this.getConstraint(conn);
+            if (constraint == null) {
+                constraint = Collections.EMPTY_LIST;
+            }
+            //may brake idk why
+            Point start = conn.getSourceAnchor().getReferencePoint().getCopy();
+            Point end = conn.getTargetAnchor().getReferencePoint().getCopy();
+            this.container.translateToRelative(start);
+            this.container.translateToRelative(end);
+            path.setStartPoint(start);
+            path.setEndPoint(end);
+            if (constraint.isEmpty()) {
+                path.setBendpoints(null);
+            } else {
+                PointList bends = new PointList(constraint.size());
+                for (Object o : constraint) {
+                    Bendpoint bp = (Bendpoint) o;
+                    bends.addPoint(bp.getLocation());
+                }
+
+                path.setBendpoints(bends);
+            }}
+
+        this.staleConnections.clear();
+    }
+
+    private void hookAll() {
+        this.figuresToBounds = new HashMap<>();
+
+        for(int i = 0; i < this.container.getChildren().size(); ++i) {
+            this.addChild(this.container.getChildren().get(i));
+        }
+
+        this.container.addLayoutListener(this.listener);
+    }
+
+    @Override
+    public void route(Connection connection) {
+        if (this.isDirty) {
```

Comment:
Please invert this condition and use an early return to avoid making lots of nesting.